### PR TITLE
dev task - Adding a new flag to *validate* to run specific validations

### PR DESCRIPTION
**Pull request by MLainer1**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47233

## Description
Adding a new flag to *validate* to run specific validations using decorators
## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -94,7 +94,8 @@ pyproject.toml
 .travis.yml
 git_push.sh
 .swagger-codegen
-.vscode
+.vscode/*
```
Comment (dantavori) [inline .gitignore:97]: remove

**Review by dantavori — CHANGES_REQUESTED** [review]

**Review by idovandijk — COMMENTED** [review]
@MLainer1 if I understood correctly the normal `validate` command remains the same right? We're just adding the ability to also run specific validations if needed

File: demisto_sdk/commands/common/hook_validations/xsoar_config_json.py [hunk 2]
```diff
@@ -26,9 +26,9 @@ class XSOARConfigJsonValidator(BaseValidator):
     """
 
     def __init__(self, configuration_file_path, json_file_path=None,
-                 ignored_errors=None, print_as_warnings=False, suppress_print=False):
+                 ignored_errors=None, print_as_warnings=False, suppress_print=False, specific_validations=None):
         super().__init__(ignored_errors=ignored_errors, print_as_warnings=print_as_warnings,
-                         suppress_print=suppress_print, json_file_path=json_file_path)
+                         suppress_print=suppress_print, json_file_path=json_file_path, specific_validations=specific_validations)
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/xsoar_config_json.py:31]: please add missing decorator in `load_xsoar_configuration_file()`

**Review by dantavori — CHANGES_REQUESTED** [review]
exhausting 😓
but great job.
please open an issue for phase 2: create a test for validate which checks that every validation function has a decorator.

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 3]
```diff
@@ -79,13 +79,14 @@ class PackUniqueFilesValidator(BaseValidator):
...
-                 skip_id_set_creation=False, prev_ver=main_branch, json_file_path=None, support=None):
+                 skip_id_set_creation=False, prev_ver=main_branch, json_file_path=None, support=None,
+                 specific_validations=None):
         """Inits the content pack validator with pack's name, pack's path, and unique files to content packs such as:
         secrets whitelist file, pack-ignore file, pack-meta file and readme file
         :param pack: content package name, which is the directory name of the pack
         """
         super().__init__(ignored_errors=ignored_errors, print_as_warnings=print_as_warnings,
-                         suppress_print=suppress_print, json_file_path=json_file_path)
+                         suppress_print=suppress_print, json_file_path=json_file_path, specific_validations=specific_validations)
```
[the remark this answers is not in the document]
Comment (MLainer1) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:89]: This command is critical for other functions in validate, so we need to run it everytime

File: demisto_sdk/commands/validate/validate_manager.py [hunk 4]
```diff
@@ -829,14 +829,20 @@ def validate_test_playbook(self, structure_validator, pack_error_ignore_list):
                                                         json_file_path=self.json_file_path)
         return test_playbook_validator.is_valid_test_playbook(validate_rn=False)
 
-    def validate_release_notes(self, file_path, added_files, modified_files, pack_error_ignore_list, is_modified):
-        pack_name = get_pack_name(file_path)
-
-        # added new RN to a new pack
+    @error_codes('RN108')
+    def validate_release_notes_for_new_pack(self, pack_name, file_path):
```
Comment (dantavori) [inline demisto_sdk/commands/validate/validate_manager.py:833]:
```suggestion
    def validate_no_release_notes_in_new_pack(self, pack_name, file_path):
```

**Review by dantavori — APPROVED** [review]
Happy Passover!
Let's do some last tests next week before you merge

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 5]
```diff
@@ -870,6 +879,7 @@ def is_not_valid_display_configuration(self):
 
         Returns:
             bool. Whether the display is there for non-hidden fields.
+            Returns True if valid, and False otherwise.
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/integration.py:882]: please change this method's name to `is_valid_display_configuration`

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 6]
```diff
@@ -781,7 +786,7 @@ def is_removed_integration_parameters(self):
             if self.handle_error(error_message, error_code, file_path=self.file_path,
                                  warning=self.structure_validator.quiet_bc):
                 self.is_valid = False
-                is_removed_parameter = True
+                is_removed_parameter = False
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/integration.py:789]: same here, make sure your method & variable names are correct (should be something like `no_removed_integration_parameters_exist` and `no_removed_parameters`)

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 7]
```diff
@@ -744,33 +747,35 @@ def is_changed_context_path(self):
         """Check if a context path as been changed.
 
         Returns:
-            bool. Whether a context path as been changed.
+            bool. True if valid, and False otherwise.
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/integration.py:750]: is_changed_context_path -> no_change_to_context_path

**Pull request merged**